qa/rpc-tests/wallet-dump.py

```diff
@@ -20,35 +21,41 @@ def read_dump(file_name, addrs, hd_master_addr_old):
         for line in inputfile:
             # only read non comment lines
             if line[0] != "#" and len(line) > 10:
-                # split out some data
-                key_label, comment = line.split("#")
-                # key = key_label.split(" ")[0]
-                keytype = key_label.split(" ")[2]
-                if len(comment) > 1:
-                    addr_keypath = comment.split(" addr=")[1]
-                    addr = addr_keypath.split(" ")[0]
-                    keypath = None
-                    if keytype == "inactivehdmaster=1":
-                        # ensure the old master is still available
-                        assert(hd_master_addr_old == addr)
-                    elif keytype == "hdmaster=1":
-                        # ensure we have generated a new hd master key
-                        assert(hd_master_addr_old != addr)
-                        hd_master_addr_ret = addr
-                    else:
-                        keypath = addr_keypath.rstrip().split("hdkeypath=")[1]
-
-                    # count key types
-                    for addrObj in addrs:
-                        if addrObj['address'] == addr and addrObj['hdkeypath'] == keypath and keytype == "label=":
-                            found_addr += 1
-                            break
-                        elif keytype == "change=1":
-                            found_addr_chg += 1
-                            break
-                        elif keytype == "reserve=1":
-                            found_addr_rsv += 1
-                            break
+
+                lparts = re.findall(r"[a-zA-Z0-9\=\/\:\-']{2,}", line)
```

Comment:
`[^#^\s]{2,}` will be more easy to read. Also IMO just `parts` instead of `lparts` look better.

Reply:
Do not agree. Cause in my regexp you can see valid symbols in dumpwallet.  And this is type of data validation. At least I exclude using of !@$...
But your regexp will match all special symbols.

Reply:
Yes that is what it should do because `importwallet` also splitting using space too (see here https://github.com/zcoinofficial/zcoin/blob/master/src/wallet/rpcdump.cpp#L461). If we follow the same format as `importwallet` that mean it will have more chance to catch bug when `dumpwallet` change it output format.